### DISCUSSION TITLE
Extracted Rust uses forbidden "loop" word as variable

### DISCUSSION THREAD

**Discussion by sethahrenbach** [AU-COBRA/ConCert#216] [Q&A]
I have found that my extracted Rust code names a variable `loop`, which is a reserved name in Rust. I'm not sure which library is generating this issue, but my Coq code does not use `loop` at the top level. Any idea what is going on here?

**Comment by 4ever2** [comment]
Without seeing your code, it is hard to say where it is coming from.
We haven't experienced this problem before with any of the code we have extracted and the word `loop` doesn't occur anywhere in the ConCert codebase. So my best guess would be that it comes from your code or a dependency of your code.

**Reply by 4ever2** [reply]
If you manage to figure out where the variable is coming from you can configure the extraction to rename it to avoid the problem. But this renaming step naturally adds to the trusted code base, so you must be careful when using it.
